keras_nlp/layers/transformer_decoder.py

```diff
@@ -272,7 +271,18 @@ def call(
         x = decoder_sequence  # Intermediate result.
 
         # Compute self attention mask.
-        self_attention_mask = compute_causal_mask(decoder_sequence)
+        batch_size = tf.shape(decoder_sequence)[0]
+        input_length = output_length = tf.shape(decoder_sequence)[1]
+        # When doing cached decoding, our attention mask is as long as the cache
```

Comment:
This looks a bit confusing to me - the input seq should be of len 1, so the attention mask is of size `[batch_size, seq_len (cache_len), 1]`. When we say "as long as the cache", it took me a while to understand what it means...

Also we may want to comment on the shape of `attention_mask`, I thought it should be `[batch_size, 1, seq_len (cache_len)]`, but running the code I found it's `[batch_size, seq_len (cache_len), 1]`.

Reply:
Yeah, I should probably clearer here. I will reword to make clear the attention mask maps from a source sequence of length `cache_length` to a dest sequence of length `1`.